### PR TITLE
Add command osdctl cluster health [OSD-7422]

### PR DESCRIPTION
**Pull request by aliceh**
In this commit I added a command that lists the expected number of nodes for a cluster and the actual number of running instances in AWS. The command outputs an error message if the cloud provider is not AWS.
Usage ' osdctl  cluster health -p [account profile] -C [internal cluster ID] '. This command requires the ocm refresh token set as an enviromental variable OCM_TOKEN.

### CODE REVIEW

**Comment by clcollins** [comment]
Your PR includes a vendor dir, but osdctl is using Go modules. That needs to be addressed.

The error, I think, is just `go fmt` in the lint finding files that need to be formatted and committed.  If you address the vendor dir, I think it's just one line in status.go:

```go
-       for idx, _ := range instances.Reservations {
+       for idx := range instances.Reservations {
```

File: cmd/cluster/status.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+		}
+	}
+
+	return nil
+}
+
+func (o *statusOptions) run() error {
+
+	// This call gets the availability zone of the cluster, as well as the numbers of compute, infra, and master nodes.
+	az, compute, infra, master, err := ocmDescribe(o.k8sclusterresourcefactory.ClusterID)
```
Comment (iamkirkbater) [inline cmd/cluster/status.go:84]: We should add an explicit check to see what the cloud provider of an cluster is, in order to explicitly fail for non-AWS clusters so SREs running this on, say, a GCP cluster won't spend time debugging why their AWS access isn't working.

File: cmd/cluster/status.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+	log.Println("Getting instances info:")
+
+	instances, err := awsJumpClient.DescribeInstances(&ec2.DescribeInstancesInput{})
+
+	totalRunning := 0
+
+	//Here we count the number of customer's running instances in the region
+	for idx := range instances.Reservations {
+		for _, inst := range instances.Reservations[idx].Instances {
+			if *inst.State.Name == "running" {
```
Comment (iamkirkbater) [inline cmd/cluster/status.go:144]: While we should also check for instances running, we should check the tags on a given instance to make sure that it's part of the cluster that we're looking at.  

With CCS we added the ability to add multiple clusters to the same account, and with BYOVPC customers could be running their own workloads on other instances within their cluster.

**Review by iamkirkbater — CHANGES_REQUESTED** [review]
While the two changes are pretty important functional and usability changes, this is a great first pass at this functionality, and I acknowledge the amount of work that was put into this and it's very appreciated.

File: cmd/cluster/status.go (hunk 1 above)
Comment (aliceh) [reply cmd/cluster/status.go:84]: The command will throw an "Error: cluster provider is not AWS", the check is on L194. I agree that the error message could be better, for example, "This command is supported only for AWS clusters. The command is not supported for <cluster provider extracted by the command> clusters."

File: cmd/cluster/status.go (hunk 2 above)
Comment (aliceh) [reply cmd/cluster/status.go:144]: Good point and good suggestion. I agree.

File: cmd/cluster/status.go (hunk 1 above)
Comment (aliceh) [reply cmd/cluster/status.go]: I added the change.

File: cmd/cluster/status.go (hunk 2 above)
Comment (aliceh) [reply cmd/cluster/status.go]: I added the change.

**Comment by iamkirkbater** [comment]
Awesome!  This looks great!

/lgtm

**Review by iamkirkbater — APPROVED** [review]

**Pull request merged**